### ISSUE TITLE
Type for type: string, format: binary

### ISSUE THREAD

**Issue by jonaskello** [drwpow/openapi-typescript#875]
I have this in oas 3.0 spec:

```
      responses:
        "200":
          description: ok
          content:
            application/pdf:
              schema:
                type: string
                format: binary
```

It generates code like this:

```ts
      readonly responses: {
        /** ok */
        readonly 200: {
          readonly content: {
            readonly "application/pdf": string; // should this really be string?
          };
        };
      };
```

My question is if the response content should be of type string in this case? I'm trying to return a `UInt8Array` in my handler which do not match the generated `string`  type. 

I think maybe the generated type should be `unknown` instead in this case? I think that is what is used for OAS2 with `type: file` (which is no longer supported in OAS3).

**Comment by jonaskello** [comment]
See also [this doc](https://swagger.io/docs/specification/describing-responses/) under the heading "Response That Returns a File" where the syntax above is included as an example.

**Comment by drwpow** [comment]
Yeah I think this is a good case for respecting `format`, especially because this will likely never be a string.

**Comment by drwpow** [comment]
After thinking about this a little, I do want to have more awareness of `format` however I don’t think there’s a clear, universal solution that would work for everyone without introducing major breaking changes.

In general I don’t feel marking a known type (`type: 'string'`) as unknown. But even if we, say, converted `format: 'binary'` into the type `string | Uint8Array`, it would still be a breaking change for people, and would have sweeping unintended effects if this applied to `$ref`s. Lastly, I think in this case `string` makes sense, because when working with HTTP streams you’d typically treat the incoming data as a string before converting it to a buffer, UintArray, etc.

For now, I think the best option would be to use the [custom formatter](https://github.com/drwpow/openapi-typescript#custom-formatter) outlined in the README. A major reason for adding this is dealing with `format`s and other metadata on schema for people to be able to develop their own custom types.
